src/app/subnets/views/VLANDetails/ConfigureDHCP/ConfigureDHCP.tsx

```diff
@@ -97,7 +97,15 @@ const ConfigureDHCP = ({ closeForm, id }: Props): JSX.Element | null => {
         then: Yup.string().required("End IP address is required"),
       }),
       gatewayIP: Yup.string(),
-      primaryRack: Yup.string(),
+      primaryRack: Yup.string().when(["enableDHCP", "dhcpType"], {
+        is: (
+          dhcpEnabled: boolean,
+          dhcpType: DHCPType.CONTROLLERS | DHCPType.RELAY
+        ) => dhcpEnabled && dhcpType === DHCPType.CONTROLLERS,
+        then: Yup.string().required(
+          "Rack controller must be selected to provide DHCP"
```

Comment:
Might be better to use the same concise format that's used in other error messages.

```suggestion
          "Primary rack is required"
```